WordPress/src/main/java/org/wordpress/android/ui/reader/services/discover/ReaderDiscoverLogic.kt

```diff
@@ -189,9 +178,16 @@ class ReaderDiscoverLogic(
             val cardJson = cardsJsonArray.getJSONObject(i)
             when (cardJson.getString(JSON_CARD_TYPE)) {
                 JSON_CARD_RECOMMENDED_BLOGS -> {
-                    val recommendedBlogsCardJson = cardJson.optJSONArray(JSON_CARD_DATA)
-                    if (recommendedBlogsCardJson.length() > 0) {
-                        simplifiedJson.put(index++, cardJson)
+                    val recommendedBlogs = parseDiscoverCardsJsonUseCase.parseRecommendedBlogsCard(cardJson)
+                    val recommendedBlogsJson = cardJson.optJSONArray(JSON_CARD_DATA)
+                    val notFollowedRecommencedBlogsJson = JSONArray()
+                    recommendedBlogs.forEachIndexed { blogIndex, blog ->
+                        if (!readerBlogTableWrapper.isSiteFollowed(blog.blogId, blog.feedId)) {
+                            notFollowedRecommencedBlogsJson.put(recommendedBlogsJson.get(blogIndex))
+                        }
+                    }
+                    if (notFollowedRecommencedBlogsJson.length() > 0) {
+                        simplifiedJson.put(index++, cardJson.put(JSON_CARD_DATA, notFollowedRecommencedBlogsJson))
```

Comment:
In the previous PR, I used this place to filter out cards with an empty list. In this PR I've added extra logic to filter out sites that are followed. As I wrote in the commit message
> It is possible that API returns already followed sites. We should filter out such sites as it doesn't make sens to recommend a site that is already known to the user. This might get fixed in the API but for now we should do this in the app.

Reply:
I will remove this as per [this comment](https://wp.me/pbMoDN-14X#comment-2341)